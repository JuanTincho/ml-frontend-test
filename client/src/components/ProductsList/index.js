import React, { Component } from 'react';
import axios from 'axios';
import PropTypes from 'prop-types';

class ProductsList extends Component {
  componentDidMount() {
    axios.get('/api/items?q=iPod').then(response => console.log(response.data));
  }

  render() {
    return <div />;
  }
}

ProductsList.propTypes = {};

export default ProductsList;
