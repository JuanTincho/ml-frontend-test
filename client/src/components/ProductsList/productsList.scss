@import "../../styles/colors.scss";

.products-list {
  background: #ffffff;
  border-radius: 4px;

  .product {
    border-bottom: 1px solid $color-gray-light;
    display: grid;
    grid-template-columns: 180px repeat(2, 1fr);
    grid-column-gap: 16px;
    padding: 16px;

    .column {
      &-1 {
        img {
          border-radius: 4px;
        }
      }

      &-2 {
        align-content: start;
        color: $color-gray-dark;
        display: grid;
        grid-row-gap: 32px;

        .price {
          font-size: 24px;
          margin-bottom: 0;

          img {
            margin-left: 10px;
          }
        }

        a {
          color: inherit;
          font-size: 18px;
          margin-top: 0;
          text-decoration: none;
        }
      }

      &-3 {
        align-self: center;
        color: $color-gray;
        font-size: 12px;
        justify-self: center;
      }
    }
  }
}
