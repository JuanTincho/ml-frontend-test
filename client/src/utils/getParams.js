const getParams = (query, value) => new URLSearchParams(query).get(value);

export default getParams;
