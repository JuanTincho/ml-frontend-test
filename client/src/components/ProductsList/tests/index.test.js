import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import mockedAxios from 'axios';
import ProductsList from '../index';
import getParams from '../../../utils/getParams';

jest.mock('../../../utils/getParams');

describe('ProductDetails', () => {
  const mockedSearchQuery = 'mockedQuery';
  const mockedLocation = { search: `search?=${mockedSearchQuery}` };
  let component;

  beforeEach(() => {
    mockedAxios.get.mockResolvedValue({
      data: {
        categories: [],
        items: [
          {
            id: '1',
            title: 'mockedTitle',
            picture: 'mockedPic',
            price: {
              currency: 'ARS',
              amount: 500,
            },
            condition: 'new',
            freeShipping: true,
            sold_quantity: 12,
            location: 'mockedLocation1',
          },
          {
            id: '2',
            title: 'mockedTitle2',
            picture: 'mockedPic2',
            price: {
              currency: 'USD',
              amount: 1000,
            },
            condition: 'used',
            freeShipping: false,
            sold_quantity: 30,
            location: 'mockedLocation2',
          },
        ],
      },
    });
    getParams.mockReturnValue(mockedSearchQuery);

    component = mount(
      <MemoryRouter>
        <ProductsList location={mockedLocation} />
      </MemoryRouter>,
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshotwith no products', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call to the service', () => {
    const expectedCall = `/api/items?q=${mockedSearchQuery}`;
    expect(mockedAxios.get).toHaveBeenCalledTimes(1);
    expect(mockedAxios.get).toHaveBeenCalledWith(expectedCall);
  });
});
