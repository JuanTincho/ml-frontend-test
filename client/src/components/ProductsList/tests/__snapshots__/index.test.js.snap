// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetails should match snapshotwith no products 1`] = `
Array [
  <div
    className="categories"
  />,
  <div
    className="products-list"
  />,
]
`;
